File: collect_app/src/main/java/org/odk/collect/android/utilities/DestroyableLifecyleOwner.java
package org.odk.collect.android.utilities;

import androidx.annotation.NonNull;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.LifecycleRegistry;

import static androidx.lifecycle.Lifecycle.Event.ON_DESTROY;
import static androidx.lifecycle.Lifecycle.Event.ON_RESUME;

public class DestroyableLifecyleOwner implements LifecycleOwner {

    private LifecycleRegistry lifecycleRegistry;

    public void start() {
        if (lifecycleRegistry == null) {
            lifecycleRegistry = new LifecycleRegistry(this);
        }

        lifecycleRegistry.handleLifecycleEvent(ON_RESUME);
    }

    public void destroy() {
        if (lifecycleRegistry != null) {
            lifecycleRegistry.handleLifecycleEvent(ON_DESTROY);
            lifecycleRegistry = null;
        }
    }

    @NonNull
    @Override
    public Lifecycle getLifecycle() {
        return lifecycleRegistry;
    }
}

File: collect_app/src/main/java/org/odk/collect/android/utilities/DestroyableLifecyleOwner.kt
package org.odk.collect.android.utilities

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.LifecycleRegistry

class DestroyableLifecyleOwner : LifecycleOwner {
    private var lifecycleRegistry: LifecycleRegistry? = null
    fun start() {
        if (lifecycleRegistry == null) {
            lifecycleRegistry = LifecycleRegistry(this)
        }
        lifecycleRegistry!!.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
    }

    fun destroy() {
        if (lifecycleRegistry != null) {
            lifecycleRegistry!!.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
            lifecycleRegistry = null
        }
    }

    override fun getLifecycle(): Lifecycle {
        return lifecycleRegistry!!
    }
}
